### PR TITLE
Fixing compatibility tests for v8

### PR DESCRIPTION
**Pull request by chatton**
## Description

A major version was not specified for `LocalhostClientFeatureReleases` so the tests for v8 were expecting the feature to not be supported.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/semverutil/semver.go [hunk 1]
```diff
@@ -25,7 +25,8 @@ func (fr FeatureReleases) IsSupported(versionStr string) bool {
 	const releasePrefix = "release-"
 	if strings.HasPrefix(versionStr, releasePrefix) {
 		versionStr = versionStr[len(releasePrefix):]
-		versionStr = strings.ReplaceAll(versionStr, "x", "0")
+		// replace x with 999 so the release version is always larger than the others in the release line.
+		versionStr = strings.ReplaceAll(versionStr, "x", "999")
```
Comment (chatton) [inline e2e/semverutil/semver.go:29]: unrelated fix to ensure semver version comparisons are accurate

**Review by crodriguezvega — APPROVED** [review]

**Review by DimitrisJim — APPROVED** [review]

File: e2e/semverutil/semver.go (hunk 1 above)
Comment (DimitrisJim) [reply e2e/semverutil/semver.go:29]: https://github.com/cosmos/ibc-go/blob/e51b60360be8305f765305e711a29bc4d85f74c3/scripts/update_compatibility_tests.py#L72-L77 prepare for the corner if wrong

Comment (chatton) [reply e2e/semverutil/semver.go:29]: I think `main` > `release-x.y.z` > `x.y.z`

`main` already gets special treatment in this function
